### PR TITLE
Strong name key pair

### PR DESCRIPTION
**Pull request by ramarag**
The coreclr side of changes for https://github.com/dotnet/corefx/issues/11808

@jkotas @weshaggard PTAL

### CODE REVIEW

**Comment by jkotas** [comment]
LGTM modulo comments.

**Comment by weshaggard** [comment]
@jkotas are you OK with this remaining in CoreLib? What do you think we will do for this in CoreRT that doesn't require a bunch of Crypto being pulled in?

**Comment by jkotas** [comment]
AssemblyName has StrongNameKeyPair property, and it is tied with the rest of the reflection -> has to be in CoreLib.

The crypto is not the ugly problem for this one: It does not show up in the public surface, and the class itself does not actually do any real crypto - it just picks a couple of bytes from the private key blob. We can not-support it (it is what I have recommended to Rama), or just do it with simplifying assumptions that the algorithm is RSA, or call the up stack crypto via reflection.

The ugly problem for this one is the FileStream in the public surface. I think we should remove it from the netstandard because of it would be very unfortunate to fold the FileStream into corelib just for this one, it is rarely used and it has trivial workaround (call the other constructor - `new StrongNamePair(File.ReadAllBytes(...))`). @ramarag Could you please file issue on it in the dotnet/standard repo?

File: src/mscorlib/src/System/Reflection/StrongNameKeyPair.cs [hunk 1]
```diff
@@ -49,6 +39,7 @@ public class StrongNameKeyPair : IDeserializationCallback, ISerializable
         private String  _keyPairContainer;
```
Comment (jkotas) [inline src/mscorlib/src/System/Reflection/StrongNameKeyPair.cs]: Actually - scratch this.

Instead, could you please add the ctor that takes string; and add the PublicKey property and implement it by throwing PNSE exception.

File: tests/src/sanity/temp.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+using System.Runtime;
+using System.Runtime.InteropServices;
+using System.Reflection;
+using System.IO;
+using System.Threading;
+using System.Collections.Generic;
+using System.Collections;
+using System.Runtime.Serialization;
+
+class Example
```
Comment (ramarag) [inline tests/src/sanity/temp.cs]: done

File: src/mscorlib/src/System/Reflection/StrongNameKeyPair.cs (hunk 1 above)
Comment (ramarag) [inline src/mscorlib/src/System/Reflection/StrongNameKeyPair.cs]: done

**Comment by ramarag** [comment]
The OSX and Linux failures are pre exisitng failures

**Pull request merged**